tests/Agent/IntegrationTests/UnboundedIntegrationTests/Elasticsearch/ElasticsearchTests.cs

```diff
@@ -33,13 +33,27 @@ protected enum ClientType
 
         const string IndexName = "flights";
 
+        protected readonly bool _syncMethodsOk;
+        const string SyncMethodSkipReason = "Synchronous methods are deprecated in latest Elastic.Clients.Elasticsearch";
+
 
         protected ElasticsearchTestsBase(TFixture fixture, ITestOutputHelper output, ClientType clientType) : base(fixture)
         {
             _fixture = fixture;
             _fixture.TestLogger = output;
             _clientType = clientType;
 
+            // non-async methods are deprecated in the latest Elastic.Clients.Elasticsearch versions
+            if (_clientType != ClientType.ElasticClients ||
+                (_fixture.GetType() != typeof(ConsoleDynamicMethodFixtureCoreLatest) && _fixture.GetType() != typeof(ConsoleDynamicMethodFixtureFWLatest)))
+            {
+                _syncMethodsOk = true;
+            }
+            else
+            {
+                _syncMethodsOk = false;
+            }
```

Comment:
It might be simpler to move `_syncMethodsOk` to a be a property on the fixture. You can set the property to false in the fixtures that should not have sync methods. Then the skippable facts can check the property on the fixture to determine if the fact should run or not.

Reply:
It's not just a matter of the fixture (latest Core or FW) but also which library we're testing (`ClientType.ElasticClients`), so I don't think this would be enough of a simplification to be worth it.  I also like keeping the special-case logic in the test class.

Reply:
The type checking that we're doing here is my biggest concern, because that's where we defeating the purpose of having an inheritance hierarchy.

It seems like the better place for that setting is within the test subclasses. In those subclasses you can set the appropriate value for syncMethodsOk to false for the couple of cases where we shouldn't test the sync methods.